src/Http/ServerRequest.php

```diff
@@ -1881,7 +1885,7 @@ public function getUri()
      * @param bool $preserveHost Whether or not the host should be retained.
      * @return static
      */
-    public function withUri(UriInterface $uri, $preserveHost = false)
+    public function withUri($uri, $preserveHost = false)
```

Comment:
Could this be kept? Also, bool addition possible?

Reply:
These methods are inherited from the PSR interface so can't add additional typehints for arguments.

Reply:
My bad, the interface method does have the `UriInterface` typehint.

Reply:
What about the 2nd arg? bool typehint?